src/serialization/serialization.rs

```diff
@@ -0,0 +1,412 @@
+extern crate byteorder;
+
+use super::{Counter, Histogram};
+use std::io::{self, Cursor, Read, Write};
+use std;
+use super::num::ToPrimitive;
+use self::byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+
+#[path = "tests.rs"]
+#[cfg(test)]
+mod tests;
+
+#[path = "benchmarks.rs"]
+#[cfg(all(test, feature = "bench_private"))]
+mod benchmarks;
+
+const V2_COOKIE_BASE: u32 = 0x1c849303;
```

Comment:
I was thinking that the file was getting a little porky, but didn't want to have a disruptive commit while you were reviewing... but now you've given me an excuse. :)